File: apple/Sources/FerrostarMapLibreUI/Theme/InstructionRowTheme.swift
import SwiftUI

public protocol InstructionRowTheme {
    
    /// The color for primary instruction.
    var instructionColor: Color { get }
    
    /// The font for primary instruction.
    var instructionFont: Font { get }
    
    /// The color for the step distance (or distance to step).
    var distanceColor: Color { get }
    
    /// The font for the step distance (or distance to step).
    var distanceFont: Font { get }
    
    /// The color of the icon.
    var iconTintColor: Color { get }
    
    /// The color of the background.
    var backgroundColor: Color { get }
}

public struct DefaultInstructionRowTheme: InstructionRowTheme {
    public var instructionColor: Color = .primary
    public var instructionFont: Font = .title
    public var distanceColor: Color = .secondary
    public var distanceFont: Font = .subheadline
    public var iconTintColor: Color = .primary
    public var backgroundColor: Color = .white // TODO: Dynamic color
}

File: android/composeui/src/main/java/com/stadiamaps/ferrostar/composeui/theme/InstructionRowTheme.kt
package com.stadiamaps.ferrostar.composeui.theme

import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle

/** Themes for instruction banner view components */
interface InstructionRowTheme {
  /** The text style for the step distance (or distance to step). */
  @get:Composable val distanceTextStyle: TextStyle
  /** The style for instruction text. */
  @get:Composable val instructionTextStyle: TextStyle
  /** The color of the icon. */
  @get:Composable val iconTintColor: Color
  /** The background color for the view. */
  @get:Composable val backgroundColor: Color
}

/** Default theme using the material theme. */
object DefaultInstructionRowTheme : InstructionRowTheme {
  override val distanceTextStyle: TextStyle
    @Composable
    get() = MaterialTheme.typography.titleLarge.merge(color = MaterialTheme.colorScheme.onSurface)

  override val instructionTextStyle: TextStyle
    @Composable
    get() =
        MaterialTheme.typography.headlineSmall.merge(
            color = MaterialTheme.colorScheme.onSurfaceVariant)

  override val iconTintColor: Color
    @Composable get() = MaterialTheme.colorScheme.onSurface

  override val backgroundColor: Color
    @Composable get() = MaterialTheme.colorScheme.surfaceContainerLow
}
